### DISCUSSION TITLE
Add a easy way to find a collection by slug path

### DISCUSSION THREAD

**Discussion by edcoreweb** [lunarphp/lunar#137] [Feature Requests]
Would be nice to have a `findByPathOrFail` , method, where you could pass in the slug path, and get pack the collection.

```php
Collection::findByPathOrFail('a/b/c/d')
```

Would this be something you are willing to add in?

**Comment by alecritson** [comment]
Currently, we fetch the `Url` model and eager load the `element` relation for this purpose. It's come up more than once, however, so I'm open to discussing this. I'll move this to the discussions area :)

**Comment by edcoreweb** [comment]
Yes, the way we do it is similar, but was looking for a cleaner way than this:

```php
$pieces = explode('/', $slug);

$collections = Url::whereElementType(Collection::class)
    ->whereIn('slug', $pieces)
    ->with(['element'])
    ->orderByRaw('FIND_IN_SET(slug, ?)', [implode(',', $pieces)])
    ->get()
    ->map
    ->element;

// ensure we have found all pieces
abort_unless($collections->count() === count($pieces), 404);

// ensure path hierarchy
// a/b/c/d
// 1/20/5/30 => null/1/20/5
$idPath = $collections->map->id->join('/');
$parentPath = $collections->map->parent_id->push($collections->last()->id)->filter()->join('/');

abort_unless($idPath === $parentPath, 404);
```
This only does 2 queries no mater how many nesting levels you have.

**Reply by alecritson** [reply]
Are you on the discord server? I think there has been a similar discussion about this on there.

https://discord.gg/v6qVWaf
